File: payments/src/main/java/br/com/bank/payments/api/exception/ExceptionsHandler.java
package br.com.bank.payments.api.exception;

import br.com.bank.payments.api.dto.output.ErrorDTO;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class ExceptionsHandler {
    @ExceptionHandler(Exception.class)
    private ResponseEntity<ErrorDTO> handleException(Exception e) {
        ErrorDTO error = new ErrorDTO(HttpStatus.BAD_REQUEST.name(), e.getMessage());
        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(error);
    }

    @ExceptionHandler(NotFoundException.class)
    private ResponseEntity<ErrorDTO> handleNotFoundException(Exception e) {
        ErrorDTO error = new ErrorDTO(HttpStatus.BAD_REQUEST.name(), e.getMessage());
        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(error);
    }
}

File: users/src/main/kotlin/br/com/bank/users/api/exception/ExceptionsHandler.kt
package br.com.bank.users.api.exception

import br.com.bank.users.api.dto.output.ErrorDTO
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class ExceptionsHandler {

    @ExceptionHandler(Exception::class)
    fun handleException(e: Exception): ResponseEntity<ErrorDTO> {
        val error = ErrorDTO(error = HttpStatus.BAD_REQUEST.name, message = e.message)
        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(error)
    }

    @ExceptionHandler(NotFoundException::class)
    fun handleNotFoundException(e: NotFoundException): ResponseEntity<ErrorDTO> {
        val errorResponse = ErrorDTO(error = HttpStatus.BAD_REQUEST.name, message = e.message)
        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(errorResponse)
    }
}